### PR TITLE
Add support for custom file extensions

### PR DESCRIPTION
**Pull request by NullVoxPopuli**

### CODE REVIEW

**Comment by rwjblue** [comment]
I'm a bit worried about this being a breaking change. Landing this would mean _existing_ codemods would now be running against non-js files (and would almost certainly error when ran).

I wonder if we can have the host codemod-cli project specifically opt-in to the extensions it supports. Either by having them pass that info in when calling `bin-support.js` (and we could even default to `ts,js`) or by specifying something in their `package.json` that we can read, or something smarter than both of those... 🤔 

What do you think?

**Comment by NullVoxPopuli** [comment]
Sounds good, check my update ;)

File: README.md [hunk 1]
```diff
@@ -57,6 +57,25 @@ can update your project's README and your transforms README via:
...
+By default the bin script that is generated for your `codemod-cli` project will run against `.js` and `.ts` files.
+If you'd like to change that (e.g. to run against `.hbs` or `.jsx` files) you can tweak your projects `bin/cli.js` script
+to add `--extensions=hbs,jsx`:
+
+```js
+#!/usr/bin/env node
+'use strict';
+
+require('codemod-cli').runTransform(
+  __dirname,
```
Comment (NullVoxPopuli) [inline README.md]: In the no-implicit-this codemod, I have this: https://github.com/ember-codemods/ember-no-implicit-this-codemod/pull/4/files

Comment (NullVoxPopuli) [reply README.md]: Does what you have also work?

File: tests/cli-test.js [hunk 2]
```diff
@@ -392,6 +393,34 @@ QUnit.module('codemod-cli', function(hooks) {
...
+            },
+          },
+        });
+
+        userProject.write({
+          foo: { 'something.hbs': `<Foo />` },
+        });
+
+        await CodemodCLI.runTransform(codemodProject.path('bin'), 'main', [
+          '--extensions',
```
Comment (NullVoxPopuli) [inline tests/cli-test.js]: Yeah, array args :)

File: README.md (hunk 1 above)
Comment (NullVoxPopuli) [reply README.md]: There are missing commas here, too

Comment (rwjblue) [reply README.md]: Good catch! Updated...

**Comment by rwjblue** [comment]
I've updated this to instead add an extra argument to the end (instead of being "inside" the `paths or globs` thing), but I'm going to do a follow up PR to migrate this to use an object (all these ordered arguments are frustrating).

I want to do that separately though so that I can ensure that existing support still works, and new projects get generated with the POJO syntax.

**Pull request merged**